mlflow/utils/_capture_modules.py

```diff
@@ -95,6 +95,49 @@ def __exit__(self, *_, **__):
         importlib.import_module = self.original_import_module
 
 
+class _CaptureImportedModulesForHF(_CaptureImportedModules):
+    """
+    A context manager to capture imported modules by temporarily applying a patch to
+    `builtins.__import__` and `importlib.import_module`.
+    Used for 'transformers' flavor only.
+    """
+
+    def __init__(self, module_to_throw):
+        super().__init__()
+        self.module_to_throw = module_to_throw
+
+    def _wrap_package(self, name):
+        if name == self.module_to_throw or name.startswith(f"{self.module_to_throw}."):
+            raise ImportError(f"Disabled package {name}")
+
+    def _wrap_importlib_util_find_spec(self, original):
+        @functools.wraps(original)
+        def wrapper(name, package=None):
+            if name == self.module_to_throw:
+                return None
+            return original(name, package)
+
+        return wrapper
+
+    def _record_imported_module(self, full_module_name):
+        self._wrap_package(full_module_name)
+        return super()._record_imported_module(full_module_name)
+
+    def __enter__(self):
+        # Patch `importlib.util.find_spec`
+        self.original_importlib_util_find_spec = importlib.util.find_spec
+        importlib.util.find_spec = self._wrap_importlib_util_find_spec(
+            self.original_importlib_util_find_spec
+        )
```

Comment:
Q:

What happens if we don't patch `importlib.util.find_spec` ? 

Because in `import_module` patching we already make it raise error when importing package name equals to `self.module_to_throw`